restler/engine/core/driver.py

```diff
@@ -344,222 +413,10 @@ def add_producers(req, req_collection, req_list, dfs_stack):
 def compute_request_goal_seq(request, req_collection):
     req_list = add_producers(request, req_collection, [], [])
     if req_list == [] :
-        logger.write_to_main(f"\nSkipping request {request.method} {request.endpoint}\n", True)
+        logger.write_to_main(f"\nSkipping request {request.method} {request.endpoint}"\
+                             f"because a sequence satisfying all dependencies was not found.\n", True)
     return req_list
 
-
-def generate_sequences_directed_smoketest(fuzzing_requests, checkers):
```

Comment:
There might be a major design flaw introduced with these changes here.

The goal of the *directed* smoke test is to provide a way to test every request *once* with a cost *linear* in the number of requests in an API. This code guarantees that, for each request, a *single* sequence leading to the target request is selected, and then the target request is executed.

By eliminating this code, we go back to a *non-linear* algorithm like the previous/old smoke test (where multiple sequences corresponding to different interleavings of the partial order of dependency relations may be executed for a same target request).

In other words, this change undoes the very purpose of introducing a *directed* smoke test in the first place: with this change, the smoke test is no longer *directed* (like before the directed smoke test was implemented).

Please revert that part of this PR.

Reply:
The code to calculate sequences for directed smoke test is moved, not deleted.  See Driver.py Line ~507.  The main algorithm then only executes these sequences, including only once the prefix for each necessary sequence.

Reply:
The difference from the previous code is the order of the sequences: before, the algorithm was "render each prefix, up until the request being covered", now the algorithm is "for each generation, render all prefixes of a particular generation that will be required for the goal request."

Reply:
Note that the directed smoke test code already sorted the requests by their depth in the dependency graph. So the exploration was also generation by generation.

Reply:
But now, it seems root-cause identification is gone, cycle detection is gone, etc.

Reply:
Sorry, my mistake: cycle detection is still there, great!

Thank you also for the other clarifications.